core/scheduler/service/job_run_service.go

```diff
@@ -801,7 +827,53 @@ func (s *JobRunService) createOperatorRun(ctx context.Context, event *scheduler.
 		s.raiseJobRunStateChangeEvent(jobRun)
 	}
 
-	return s.operatorRunRepo.CreateOperatorRun(ctx, event.OperatorName, operatorType, jobRun.ID, event.EventTime)
+	job, err := s.jobRepo.GetJob(ctx, event.Tenant.ProjectName(), event.JobName)
+	if err != nil {
+		s.l.Error("error getting job by job name [%s]: %s", event.JobName, err)
+		return err
+	}
+
+	existingOperatorRunState, err := s.operatorRunRepo.GetOperatorRunStateIfExist(ctx, event.OperatorName, operatorType, jobRun.ID)
+	if err != nil {
+		s.l.Error("error checking existing operator run state [%s], operatorName [%s] : %s", jobRun.ID, event.OperatorName, err)
+		return err
+	}
+
+	err = s.operatorRunRepo.CreateOperatorRun(ctx, event.OperatorName, operatorType, jobRun.ID, event.EventTime)
+	if err != nil {
+		s.l.Error("error registering operator run job [%s], type [%s], operator [%s] : %s", event.JobName, operatorType, event.OperatorName, err)
+		return err
+	}
+
+	if existingOperatorRunState == nil || *existingOperatorRunState != scheduler.StateRetry {
```

Comment:
In my opinion we should see this whole operation as two sub functions fetchAllSLAAlertConfigs & Registration.